cli/tests/activate.bats

```diff
@@ -984,6 +984,83 @@ EOF
   assert_output --partial "test_alias is an alias for echo testing"
 }
 
+# bats test_tags=activate,activate:rc:zsh
+@test "zsh: interactive activate respects history settings from dotfile" {
+  project_setup
+
+  # This should always work, even when Darwin sets a default in `/etc/zshrc`.
+  echo 'HISTFILE=${PROJECT_DIR}/.alt_history' >"$HOME/.zshrc.extra"
+  echo 'SHELL_SESSION_DIR=${PROJECT_DIR}/.alt_sessions' >>"$HOME/.zshrc.extra"
+
+  # TODO: flox will set HOME if it doesn't match the home of the user with
+  # current euid. I'm not sure if we should change that, but for now just set
+  # USER to REAL_USER.
+  FLOX_SHELL="zsh" USER="$REAL_USER" NO_COLOR=1 \
+    run expect "$TESTS_DIR/activate/histfile.exp" "$PROJECT_DIR"
+  assert_success
+  assert_line --partial "HISTFILE=$PROJECT_DIR/.alt_history"
+  assert_line --partial "SHELL_SESSION_DIR=$PROJECT_DIR/.alt_sessions"
+}
+
+# bats test_tags=activate,activate:rc:zsh
+@test "zsh: interactive activate respects history settings from dotfile based on original ZDOTDIR" {
+  project_setup
+
+  # Mimic the default `/etc/zshrc` on Darwin prior to Nix being installed. We
+  # have to do this in `~/.zshrc` because we can't mock `/etc/zshrc`. However we
+  # apply the same `ZDOTDIR` logic to both, ensuring that it's not pointing at
+  # our immutable storepath.
+  echo 'HISTFILE=${ZDOTDIR:-$HOME}/.alt_history' >"$HOME/.zshrc.extra"
+  echo 'SHELL_SESSION_DIR=${ZDOTDIR:-$HOME}/.alt_sessions' >>"$HOME/.zshrc.extra"
+
+  # TODO: flox will set HOME if it doesn't match the home of the user with
+  # current euid. I'm not sure if we should change that, but for now just set
+  # USER to REAL_USER.
+  FLOX_SHELL="zsh" USER="$REAL_USER" NO_COLOR=1 \
+    run expect "$TESTS_DIR/activate/histfile.exp" "$PROJECT_DIR"
+  assert_success
+  assert_line --partial "HISTFILE=$HOME/.alt_history"
+  assert_line --partial "SHELL_SESSION_DIR=$HOME/.alt_sessions"
+}
+
+# bats test_tags=activate,activate:rc:zsh
+@test "zsh: interactive activate respects history settings from environment variable where available" {
+  project_setup
+
+  # TODO: flox will set HOME if it doesn't match the home of the user with
+  # current euid. I'm not sure if we should change that, but for now just set
+  # USER to REAL_USER.
```

Comment:
I answered that somewhere ... that is required for use with `sudo flox`, because Nix writes to `HOME`. Users could instead run `sudo -H flox`, but that's an annoying requirement to place on a user after they've invoked flox. We could wrap the setting of `HOME` in a test of `SUDO_USER` or something like that, but in any case that's the reason for the logic. Can we update this comment to reflect that we are _not_ unsure (forgive the double negative!). Thanks.

Reply:
This is a copy/paste from several other tests in this file:
```
% rg --count 'TODO: flox will set HOME'
cli/tests/activate.bats:9
```

I want to get this PR merged ASAP because it's carried over from the previous sprint but I can follow-up on all of the comments in a subsequent PR if you think it's important.